src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java

```diff
@@ -339,34 +330,20 @@
 	 * @param clazz
 	 * @return
 	 */
-	<T> long count(CriteriaQuery query, Class<T> clazz);
+	<T> long count(Query query, @Nullable Class<T> clazz);
```

Comment:
`Query` currently carries types and indices. Typically, we keep these table-like coordinates on method level so a `Query` object can be applied to various (tables/collections/)indices. I wonder whether it would make sense to extract indices/types into a type that holds both information. We could then add these coordinates to `count(…)` via `count(Query, Class<?>, Index...)`. That way, we separate the actual criteria from execution-specific targets.

Ideally, we can expose `count` with the following overloads:

* `long count(Query query, Class<?> entityClass);`
* `long count(Query query, Index... indexes);`
* `long count(Query query, @Nullable Class<?> entityClass, Index... indexes);`

We should also drop the generic type `T` as there's no outbound usage of `T`.